File: services/graph4j_service/src/network/envelope.py
import os


class Envelope:
    @staticmethod
    def create_end_user_envelope(socket_id, event_name):
        library_name = os.environ["LIBRARY_NAME"]

        return {
            "header": {
                "socket_id": socket_id,
                "event_name": event_name,
                "library_name": library_name
            },
            "payload": {}
        }

    @staticmethod
    def create_selector_envelope(file_id, algorithm_name):
        library_name = os.environ["LIBRARY_NAME"]

        return {
            "header": {
                "file_id": file_id,
                "algorithm_name": algorithm_name,
                "library_name": library_name
            },
            "payload": {}
        }

File: services/graph4j_service/src/process/src/main/java/process/network/Envelope.java
package process.network;

import org.json.JSONObject;

public class Envelope {
    private static final String libraryName = System.getenv("LIBRARY_NAME");

    public static String createSelectorEnvelope(String fileId, String algorithmName, String result) {
        return new JSONObject()
                .put("header",
                        new JSONObject()
                                .put("file_id", fileId)
                                .put("algorithm_name", algorithmName)
                                .put("library_name", libraryName)
                )
                .put("payload",
                        new JSONObject()
                                .put("result", result)
                ).toString();
    }
}
